pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -596,20 +596,20 @@ class Timestamp(_Timestamp):
             return Timestamp(datetime(year, month, day, hour or 0,
                                       minute or 0, second or 0,
                                       microsecond or 0, tzinfo),
-                             tz=tz)
+                             nanosecond=nanosecond, tz=tz)
         elif is_integer_object(freq):
             # User passed positional arguments:
             # Timestamp(year, month, day[, hour[, minute[, second[,
             # microsecond[, tzinfo]]]]])
             return Timestamp(datetime(ts_input, freq, tz, unit or 0,
                                       year or 0, month or 0, day or 0,
-                                      hour), tz=hour)
+                                      hour), nanosecond=minute, tz=hour)
 
         if tzinfo is not None:
             # User passed tzinfo instead of tz; avoid silently ignoring
             tz, tzinfo = tzinfo, None
 
-        ts = convert_to_tsobject(ts_input, tz, unit, 0, 0)
+        ts = convert_to_tsobject(ts_input, tz, unit, 0, 0, nanosecond or 0)
```

Comment:
What if we get here with a string ts_input with conflicting nanos? I think it might be better to accept nanos _only_ in combination with other integer args/kwargs.

Reply:
Currently (and in this PR), the Timestamp constructor will just parse the string and ignore all additional passed kwargs related to the values of each datetime element:

```
In [7]: Timestamp('2010-10-10 12:59:59.999999999', hour=0, nanosecond=16)
Out[7]: Timestamp('2010-10-10 12:59:59.999999999')
``` 

The nanosecond kwarg will only be used if a datetime.datetime is passed or other datetime element kwargs are passed. As mentioned in my other comment, we might also want to raise in the situation above.

Reply:
I agree here, the nanoscecond kw must only be used when kwargs are passed